bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -118,8 +131,42 @@ bugsnag_event *bsg_event_read(int fd) {
   } else if (event_version == 3) {
     bugsnag_report_v3 *report_v3 = bsg_report_v3_read(fd);
     event = bsg_map_v3_to_report(report_v3);
-  } else {
-    event = bsg_report_v4_read(fd);
+  } else if (event_version == 4) {
+    bugsnag_report_v4 *report_v4 = bsg_report_v4_read(fd);
+    event = bsg_map_v4_to_report(report_v4);
+  } else if (event_version == 5) {
+    event = bsg_report_v5_read(fd);
+  }
+  return event;
+}
+
+bugsnag_event *bsg_map_v4_to_report(bugsnag_report_v4 *report_v4) {
+  if (report_v4 == NULL) {
+    return NULL;
+  }
+  bugsnag_event *event = malloc(sizeof(bugsnag_event));
+
+  if (event != NULL) {
+    event->notifier = report_v4->notifier;
+    event->device = report_v4->device;
+    event->user = report_v4->user;
+    event->error = report_v4->error;
+    event->metadata = report_v4->metadata;
+    event->crumb_count = report_v4->crumb_count;
+    event->crumb_first_index = report_v4->crumb_first_index;
+    memcpy(event->breadcrumbs, report_v4->breadcrumbs,
+           sizeof(event->breadcrumbs));
+    event->severity = report_v4->severity;
+    strcpy(event->session_id, report_v4->session_id);
+    strcpy(event->session_start, report_v4->session_start);
```

Comment:
If we ever shorten the size of any `event->xyz`, these `strcpy` calls will overwrite memory.

Reply:
Updated to use `bsg_strncpy_safe` instead